File: Greedy/135. Candy.cpp
class Solution {
public:
    int candy(vector<int>& ratings) {
        int n = ratings.size();
        if (n<2)
            return n;
        vector<int> num(n, 1);
        for (int i = 1; i < n; ++i){
            if (ratings[i-1] < ratings[i]){
                num[i] = num[i-1] + 1;
            }
        }
        for (int i = n-1; i > 0; --i){
            if (ratings[i-1] > ratings[i]){
                num[i-1] = max(num[i-1], num[i] + 1);
            }
        }
        return accumulate(num.begin(), num.end(), 0);
    }
};

File: Greedy/135. Candy.py
class Solution:
    def candy(self, ratings: List[int]) -> int:
        size = len(ratings)
        if size == 0:
            return 0
        num = [1] * size
        for i in range(1, size):
            if ratings[i-1] < ratings[i]:
                num[i] = num[i-1] + 1
        for i in range(size - 1, 0, -1):
            if ratings[i-1] > ratings[i]:
                num[i-1] = max(num[i-1], num[i] + 1)
        return sum(num)